Update go DistTask.yml to support  golang 1.25.2 and Windows/ARM64 build.

cmaglie opened the pull request:
The newer Docker-ized cross-compilers require QEMU on the CI runner, so I added the corresponding installation step to the build workflow.

For reference, see: https://github.com/arduino/serial-discovery/pull/93 and https://github.com/arduino/mdns-discovery/pull/81

cmaglie committed:
Update DistTask go version to 1.25.2

cmaglie committed:
Added Windows ARM64 build support

cmaglie committed:
New docker crossbuild images requires qemu on non-arm CI runners

per1234 reviewed (changes requested):
Please also update the "Publish Tester Build" workflow:

https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/publish-go-tester-task.yml

per1234 commented on workflow-templates/assets/release-go-task/DistTasks.yml:
```diff
@@ -152,9 +178,9 @@ tasks:
     vars:
       PLATFORM_DIR: "{{.PROJECT_NAME}}_linux_arm_7"
-      BUILD_COMMAND: "go build -o {{.DIST_DIR}}/{{.PLATFORM_DIR}}/{{.PROJECT_NAME}} {{.LDFLAGS}}"
+      BUILD_COMMAND: "go build -buildvcs=false -o {{.DIST_DIR}}/{{.PLATFORM_DIR}}/{{.PROJECT_NAME}} {{.LDFLAGS}}"
```
I see you previously used a different solution to https://github.com/elastic/golang-crossbuild/issues/232:

https://github.com/arduino/arduino-cli/pull/2103 / https://github.com/arduino/arduino-cli/commit/5a5ae94620478b602b4e75092d4e71e48b5c1862#diff-d3d207494c3cad685790f3fcc5c393f06c44eb99e3e904fdf4f4e668aaaf77c1R209

https://github.com/arduino/arduino-lint/pull/761 / https://github.com/arduino/arduino-lint/commit/c4e16c9650b0579689f3feb288986133a972f046#diff-d3d207494c3cad685790f3fcc5c393f06c44eb99e3e904fdf4f4e668aaaf77c1R209

I adopted that solution in Arduino Language Server:

https://github.com/arduino/arduino-language-server/pull/201

Is there a reason you have switched to the `-buildvcs=false` approach (apparently the change/inconsistency in approach began at https://github.com/arduino/arduino-cli/pull/2752)? At first glance, the `-buildvcs=false` approach appears inferior in that the [embedded version control information](https://go.dev/doc/go1.18#go-version) seems like it could be useful in some cases (keeping in mind these assets might be used in applications where a version information capability has not been implemented in the codebase).

per1234 commented on workflow-templates/release-go-task.yml:
```diff
@@ -66,6 +68,9 @@ jobs:
           go-version-file: go.mod
 
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
```
Do you think it would be beneficial to only use QEMU when actually required, for the sake of efficiency?

Something like this:

```diff
diff --git a/workflow-templates/release-go-task.yml b/workflow-templates/release-go-task.yml
index 8014516..8c11726 100644
--- a/workflow-templates/release-go-task.yml
+++ b/workflow-templates/release-go-task.yml
@@ -28,24 +28,34 @@ jobs:
         os:
           - task: Windows_32bit
             artifact-suffix: Windows_32bit
+            use-qemu: false
           - task: Windows_64bit
             artifact-suffix: Windows_64bit
+            use-qemu: false
           - task: Windows_ARM64
             artifact-suffix: Windows_ARM64
+            use-qemu: true
           - task: Linux_32bit
             artifact-suffix: Linux_32bit
+            use-qemu: false
           - task: Linux_64bit
             artifact-suffix: Linux_64bit
+            use-qemu: false
           - task: Linux_ARMv6
             artifact-suffix: Linux_ARMv6
+            use-qemu: true
           - task: Linux_ARMv7
             artifact-suffix: Linux_ARMv7
+            use-qemu: true
           - task: Linux_ARM64
             artifact-suffix: Linux_ARM64
+            use-qemu: true
           - task: macOS_64bit
             artifact-suffix: macOS_64bit
+            use-qemu: false
           - task: macOS_ARM64
             artifact-suffix: macOS_ARM64
+            use-qemu: true
 
     steps:
       - name: Checkout repository
@@ -69,6 +79,7 @@ jobs:
           go-version-file: go.mod
 
       - name: Setup QEMU
+        if: matrix.os.use-qemu == true
         uses: docker/setup-qemu-action@v3
 
       - name: Build
```

per1234 reviewed (changes requested):
Please also update the "Publish Nightly Build" workflow:

https://github.com/arduino/tooling-project-assets/blob/update-go-DistTask/workflow-templates/publish-go-nightly-task.yml

per1234 reviewed (changes requested):
Please add the Windows ARM64 download links to the tables in the installation guide template:

https://github.com/arduino/tooling-project-assets/blob/main/other/installation-script/installation.md#latest-release

https://github.com/arduino/tooling-project-assets/blob/main/other/installation-script/installation.md#nightly-builds

per1234 commented on workflow-templates/release-go-task.yml:
```diff
@@ -66,6 +68,9 @@ jobs:
         with:
           go-version-file: go.mod
 
+      - name: Setup QEMU
```
Please evaluate using the alternative approach I proposed at https://github.com/arduino/serial-discovery/pull/94

per1234 reviewed (changes requested):
Since we can consider 1.25.2 to now be the standard Go version for use in all tooling projects, please bump this project's Go version:

https://github.com/arduino/tooling-project-assets/blob/main/go.mod#L3

cmaglie committed:
Made CGO_ENABLED in DistTask.yml configurable.

cmaglie committed:
Use safe.directory instead of disabling go' vcs stamping

cmaglie committed:
Do not depend on QEMU.

cmaglie committed:
Added Win/ARM64 build on publish-go-tester-task.yml

cmaglie committed:
Updated publish-go-nightly-task.yml

cmaglie committed:
Added Windows/ARM64 release in installation.md

cmaglie committed:
updated go version to 1.25.2

cmaglie committed:
Allow v prefix in releases.

cmaglie committed:
Make the release package archives with a root folder.

This change also allow to create the sub-folders in the `dist` directory
owned by the current user instead of `root` (since the folder is made
outside of the docker image).

cmaglie committed:
Updates to DistTask.yml
